src/ert/config/external_ert_script.py

```diff
@@ -20,22 +24,39 @@ def __init__(self, ert: EnKFMain, storage: StorageAccessor, executable: str):
         self.__job: Optional[Popen[bytes]] = None
 
     def run(self, *args: Any) -> None:
-        command = [self.__executable]
-        command.extend([str(arg) for arg in args])
+        script_content = (
+            Path(self.__executable).read_text(encoding="utf-8").splitlines()
+        )
+        tmp_script_path: Optional[str] = None
+        if self.__executable[-2:] == "sh":
```

Comment:
Is this a file extension like  "failing_script.sh" that you are testing, or is it /bin/bash or /bin/sh? Testing on file extension is not robust enough.

Reply:
It is file extension. Should I check the shebang instead? It could be `/bin/env sh`, `/bin/env bash`, `/bin/bash`, `/bin/sh` or any other shell I guess. Maybe I can check that it is not python instead, but that could also be a lot of different shebangs.

Reply:
What about checking `if "sh" in SHEBANG`?

Reply:
testing on the shebang is maybe sufficient yes. I would try to make a regexp  requiring either a space or a slash, then an optional "ba" or "c" or "tc", and then "sh" at the end.